plans/2026-07-18_opus_modernization_plan.md

```
@@ -3091,3 +3091,266 @@ body; never rewrite or delete earlier notes.*
+    unconfirmed and the `VALUES(col)` form is unchanged.** The instruction stands, and
+    its owner is whoever holds it after PR-22 establishes the server version -- it is not
+    a PR-12 omission.
+  - **PR-12a (new, assigned by the orchestrator on 2026-08-24): `_edit_cart_range`'s
+    `removerange` DELETE is not scoped to the session, and deletes other users' cart
+    rows.** Found by CodeRabbit on PR-12; given its own PR on the **PR-03a precedent**
+    (rev 7.5), because PR-12's contract is byte-identical golden responses and this fix
+    is deliberately *not* byte-identical in effect. Everything a fresh executor needs:
+    1. **The statement.** `cart/views.py`, the `elif action == 'removerange':` branch of
+       `_edit_cart_range` (the `recycle_bin == 0` path): `DELETE cart FROM cart INNER
+       JOIN <user_query_table> ON <uqt>.id=cart.obs_general_id WHERE <uqt>.sort_order
+       >= <min> AND <uqt>.sort_order <= <max>`. It names no `session_id`, so it deletes
```

markdownlint-cli2 (0.23.2)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)